src/main.js

```diff
@@ -196,8 +230,19 @@ global.webpplEval = function(s, k, a, code, runner) {
   if (runner === undefined) {
     runner = util.runningInBrowser() ? 'web' : 'cli'
   }
-  var compiledCode = compile(code);
-  return eval.call(global, compiledCode)(util.trampolineRunners[runner])(s, k, a);
+  var codeWithMap = compile(code, {filename: 'webppl:eval'});
+  try {
+
+    // Generally CPS and exceptions don't mix very well. However, I
+    // think catching errors will work here because the eval'd code is
+    // run in its own trampoline, and therefore within this try/catch.
+    // In contrast, if the eval'd code ran in the top-level trampoline
+    // this approach wouldn't work.
```

Comment:
We could make the pausy runner take an error-handling function as an argument (this would also simplify cross-browser error handling a little, see probmods/webppl-editor#9)